src/paperoni/__main__.py

```diff
@@ -124,24 +134,118 @@ def run(self):
+            for pinfo in self.iterate():
+                if ex and pinfo.key in ex:
+                    continue
+                scored = Scored(focuses.score(pinfo), PaperWorkingSet.make(pinfo))
+                top.add(scored)
+            dump(Top[Scored[PaperWorkingSet]], top, dest=work.state)
+
+    @dataclass
+    class View:
+        """View the articles in the workset."""
+
+        # Output format
+        format: Formatter = TerminalFormatter
+
+        def run(self, work):
+            top = deserialize(Top[Scored[PaperWorkingSet]], work.state)
+            papers = list(ws.value.current for ws in top)
+            self.format(papers)
+
+    # Command
+    command: TaggedUnion[Get, View]
+
+    # File containing the working set
+    # [alias: -w]
+    state: Path
+
+    # List of focuses
+    # [alias: -f]
+    focuses: Path
+
+    # Exclusion list
+    # [alias: -x]
+    exclusions: Path = None
+
+    # Number of papers to keep in the working set
+    n: int = 10
+
+    def run(self):
+        self.command.run(self)
+
+
 @dataclass
 class PaperoniInterface:
     """Paper database"""
 
-    command: TaggedUnion[Discover, Refine, Fulltext]
+    command: TaggedUnion[Discover, Refine, Fulltext, Work]
 
     def run(self):
         self.command.run()
 
 
 def main():
-    parser = argparse.ArgumentParser(add_help=False)
-    parser.add_argument("--config", default=None)
-    args, remaining = parser.parse_known_args()
-    if args.config:
-        add_overlay(Path(args.config))
-    command = cli(field="paperoni.cli", type=PaperoniInterface, argv=remaining)
-    command.run()
+    @outsight.add
+    async def show_progress(sent, dash):
+        async for name, sofar, total in sent["progress"]:
+            dash.add_progress(name, sofar, total)
+
+    @outsight.add
+    async def show_paper_stats(sent, dash):
+        async for group in sent["discover"].roll(5, partial=True):
+            values = [f"{pinfo.paper.title}" for pinfo in group]
+            dash["titles"] = History(values)
+
+    @outsight.add
+    async def show_requests(sent, dash):
+        async for group in sent["url", "params", "response"].roll(5, partial=True):
+            values = [
+                f"[{resp.status_code}] {req} {params or ''}"
+                for req, params, resp in group
+            ]
+            dash["request"] = History(values)
+
+    @outsight.add
+    async def show_score_stats(sent, dash):
+        min_score = None
+        max_score = None
+        count = 0
+        async for score in sent["score"]:
+            if score == 0:
+                continue
+            count += 1
+            if min_score is None or score < min_score:
+                min_score = score
+            if max_score is None or score > max_score:
+                max_score = score
+            dash["score"] = (
+                f"{score}   [bold green]max: {max_score}[/bold green]   [bold blue]count: {count}[/bold blue]"
+            )
```

Comment:
I wonder if there could be conflicts if two sources end up being called with the same send signature while not containing the same data? Althought, if they have the same signature chances are the same dashboard should do for these sources

Reply:
Ah, in that sense. Yeah. There may be more robust approaches, such as dedicated streams (e.g. `scores = EventStream(); scores.send(score)` and then using it as a fixture). But I feel it's not likely to be an issue within the project.